### PR TITLE
Fixes #8882: Use system variables to configure postgres db name and user for rsyslog

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8882

### CODE REVIEW

File: techniques/system/distributePolicy/1.0/rudder-rsyslog-root.st [hunk 1]
```diff
@@ -61,7 +61,7 @@ $template RudderReportsFormat,"insert into RudderSysEvents (executionDate, nodeI
 # The report format is @@Policy@@State@@RuleId@@DirectiveId@@VersionId@@Component@@Key@@ExecutionTimeStamp##NodeId@#HumanReadableMessage
 #
 # 1 - Send every matching report in the database...
-:msg, ereregex, "R: @@[ a-zA-Z0-9_\-]+?@@[a-zA-Z0-9_\-]{1,64}?@@[a-zA-Z0-9\-]+@@[a-zA-Z0-9\-]+?@@[0-9]+?@@.*?@@.*?@@[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}[+-][0-9]{1,2}:[0-9]{2}##[a-zA-Z0-9\-]+?@#.*"  :ompgsql:${install_rsyslogd.rudder_postgres_server},rudder,rudder,${p.psql_password[2]};RudderReportsFormat
+:msg, ereregex, "R: @@[ a-zA-Z0-9_\-]+?@@[a-zA-Z0-9_\-]{1,64}?@@[a-zA-Z0-9\-]+@@[a-zA-Z0-9\-]+?@@[0-9]+?@@.*?@@.*?@@[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}[+-][0-9]{1,2}:[0-9]{2}##[a-zA-Z0-9\-]+?@#.*"  :ompgsql:${install_rsyslogd.rudder_postgres_server},&RUDDER_REPORTS_DB_NAME&,&RUDDER_REPORTS_DB_USER&,${p.psql_password[2]};RudderReportsFormat
```
Comment (jooooooon) [inline techniques/system/distributePolicy/1.0/rudder-rsyslog-root.st]: In master, this file is used as-is - that is, StringTemplate in it is not parsed.

Could you add these variables to the install_rsyslogd bundle, like the others around them, and just use CFEngine variables here please?

Comment (jooooooon) [inline techniques/system/distributePolicy/1.0/rudder-rsyslog-root.st]: These variables also need to be added to the metadata.xml file

**Comment by amousset** [comment]
Commit modified

File: techniques/system/distributePolicy/1.0/rsyslogConf.st [hunk 2]
```diff
@@ -24,6 +24,8 @@ bundle agent install_rsyslogd {
 
     root_server|role_rudder_relay_top::
       "rsyslog_source_file" string => "rudder-rsyslog-root.conf";
+      "rudder-reports_db_name" string => "&RUDDER_REPORTS_DB_NAME&";
+      "rudder-reports_db_user" string => "&RUDDER_REPORTS_DB_USER&";
```
Comment (jooooooon) [inline techniques/system/distributePolicy/1.0/rsyslogConf.st]: This variable name has a dash in it - use an underscore instead please, that's not a valid CFEngine name

**Comment by amousset** [comment]
Commit modified

File: techniques/system/distributePolicy/1.0/rsyslogConf.st (hunk 2 above)
Comment (amousset) [inline techniques/system/distributePolicy/1.0/rsyslogConf.st]: indeed...

**Pull request merged**